vyper/ast/nodes.py

```diff
@@ -78,6 +79,10 @@ def get_node(
         else:
             ast_struct["ast_type"] = "VariableDecl"
 
+    if ast_struct["ast_type"] == "EnumDef":
+        warnings.warn("enum will be deprecated in a future release, use flag instead")
```

Comment:
if possible, it might be useful to also include the source location in the warning message

Reply:
Should I just mention the line at which `enum` was used in the warning or is there a function to have some pretty print the warning like when a syntax exception is raised?

Reply:
> Should I just mention the line at which `enum` was used in the warning or is there a function to pretty print the warning like when a syntax exception is raised?

there's no obvious function for this -- i think something like `str(VyperException("", node))` will work though. we should factor out the pretty-print functionality inside of VyperException but the above method works well enough for now.